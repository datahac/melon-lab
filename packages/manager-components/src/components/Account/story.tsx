import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Account from './index';

const data = {
  associatedFund: '0x0890b03f83629B397F6E5E5075400fe6Ddb4255F',
  networkId: '42',
  isCompetition: false,
  deleteWallet: action('deleteWallet'),
};

storiesOf('Components|Account', module)
  .add('Default', () => {
    return (
      <Account
        {...data}
        currentAddress="0x270c65cb31d037E4269dC8F736350FDc8ED07353"
      />
    );
  })
  .add('No Data', () => {
    return <Account {...data} />;
  });
