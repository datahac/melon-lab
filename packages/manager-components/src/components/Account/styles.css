.account {
  max-width: 400px;
  margin: 0 auto var(--space-xl) auto;
}

.account__loading {
  margin: var(--space-s) auto;
  text-align: center;
}

.account__wallet {
  background-color: var(--color-border);
  padding: var(--space-xxs) var(--space-m);
}
