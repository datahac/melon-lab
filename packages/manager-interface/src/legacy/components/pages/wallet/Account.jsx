import React from 'react';
import Account from '../../../containers/wallet/Account';

const MyAccount = () => <Account />;

export default MyAccount;
