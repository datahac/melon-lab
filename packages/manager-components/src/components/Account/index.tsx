import React, { Fragment, StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Notification from '~/blocks/Notification';
import StyledLink from '~/blocks/Link';
import Spinner from '~/blocks/Spinner';
import Link from '~/link';

import styles from './styles.css';

export interface AccountProps {
  associatedFund?: string;
  currentAddress?: string;
  deleteWallet: () => void;
  isCompetition?: boolean;
  networkId: string;
  hasAccount?: boolean;
  hasWallet?: boolean;
  loading?: boolean;
}

export const Account: StatelessComponent<AccountProps> = ({
  associatedFund,
  currentAddress,
  deleteWallet,
  isCompetition,
  networkId,
  hasAccount,
  hasWallet,
  loading,
}) => {
  const isDanger = currentAddress ? 'danger' : 'secondary';

  return (
    <div className="account">
      <style jsx>{styles}</style>
      <h3>Your Wallet</h3>
      {loading ? (
        <Spinner icon size="small" />
      ) : (
        <Fragment>
          {currentAddress ? (
            <Fragment>
              <p>
                Your ethereum address. Use this for white listing on{' '}
                <a href="https://ico.bitcoinsuisse.ch/" target="_blank">
                  ico.bitcoinsuisse.ch
                </a>
                :
                <strong>
                  <a
                    href={`https://${
                      networkId === '42' ? 'kovan.' : ''
                    }etherscan.io/address/${currentAddress}`}
                    target="_blank"
                  >
                    {' '}
                    {currentAddress}{' '}
                  </a>
                </strong>
              </p>
              {associatedFund && (
                <p>
                  Associated fund address:{' '}
                  <strong>
                    <a
                      href={`https://${
                        networkId === '42' ? 'kovan.' : ''
                      }etherscan.io/address/${associatedFund}`}
                      target="_blank"
                    >
                      {associatedFund}
                    </a>
                  </strong>
                </p>
              )}

              <p>
                <strong>
                  It is highly recommended to download a backup of your wallet.
                  You can import this into{' '}
                  <a href="https://mycrypto.com/" target="_blank">
                    MyCrypto.com
                  </a>{' '}
                  or Parity.
                </strong>
              </p>
              <p>
                <Link href="/wallet/download" passHref>
                  <StyledLink style="secondary" size="medium">
                    Download wallet backup JSON
                  </StyledLink>
                </Link>
              </p>
              {!associatedFund ? (
                <p>
                  <Link href="/setup" passHref>
                    <StyledLink style="secondary" size="medium" passHref>
                      Setup your fund
                    </StyledLink>
                  </Link>
                </p>
              ) : (
                <p>
                  <Link
                    href={{
                      pathname: '/manage',
                      query: { address: associatedFund },
                    }}
                  >
                    <StyledLink style="secondary" size="medium" passHref>
                      Go to your fund
                    </StyledLink>
                  </Link>
                </p>
              )}
              <br />
              {!isCompetition && (
                <Fragment>
                  <p>
                    <strong> [IMPORTANT] - Please read carefully</strong>{' '}
                  </p>
                  <p>
                    Careful, below actions have <strong> irreversible</strong>{' '}
                    effects. If you do not have a backup of the mnemonic phrase
                    that generated your current address,
                    <strong>
                      {' '}
                      you will never be able to access your current wallet again{' '}
                    </strong>{' '}
                    after performing one of the below actions.
                  </p>
                  <p>
                    If you do not wish to continue,{' '}
                    <Link href="/">
                      <a href="/">
                        click here to go back to your fund&#39;s page
                      </a>
                    </Link>
                    .
                  </p>
                </Fragment>
              )}
            </Fragment>
          ) : (
            <Fragment>
              <p>
                Before you can setup your fund, you need to import, restore or
                create a wallet:
              </p>
            </Fragment>
          )}

          <Fragment>
            {!hasAccount &&
              hasWallet && (
                <Fragment>
                  <Notification isWarning>
                    <p>
                      We found a Wallet on your Computer. You can load it here:
                    </p>
                    <Link href="/wallet/load" passHref>
                      <StyledLink style="secondary" size="medium">
                        Load Wallet
                      </StyledLink>
                    </Link>
                  </Notification>

                  <hr />
                </Fragment>
              )}

            {!isCompetition && (
              <p>
                <Link href="/wallet/generate" passHref>
                  <StyledLink style="secondary" size="medium">
                    Create a new wallet
                  </StyledLink>
                </Link>
              </p>
            )}
            <p>
              <Link href="/wallet/restore" passHref>
                <StyledLink style="secondary" size="medium">
                  Restore from mnemonic
                </StyledLink>
              </Link>
            </p>
            <p>
              <Link href="/wallet/import" passHref>
                <StyledLink style="secondary" size="medium">
                  Import wallet JSON
                </StyledLink>
              </Link>
            </p>
            {currentAddress &&
              !isCompetition && (
                <p>
                  <Button style={isDanger} onClick={deleteWallet}>
                    Delete wallet
                  </Button>
                </p>
              )}
          </Fragment>
        </Fragment>
      )}
    </div>
  );
};

export default Account;
