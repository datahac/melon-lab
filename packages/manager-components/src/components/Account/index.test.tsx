import React from 'react';
import Account from './index';

const mockCallbackFund = jest.fn();
const data = {
  currentAddress: '0x270c65cb31d037E4269dC8F736350FDc8ED07353',
  associatedFund: '0x0890b03f83629B397F6E5E5075400fe6Ddb4255F',
  networkId: '42',
  isCompetition: false,
  deleteWallet: () => null,
  gotoAccountGenerate: () => null,
  gotoAccountRestore: () => null,
  gotoImportJSON: () => null,
  gotoSetup: () => null,
  downloadJSON: () => null,
  goToFund: mockCallbackFund,
};

describe('Account', () => {
  const defaultElement = <Account {...data} />;
  let customeElement;

  it('should render correctly without currentAddress', () => {
    customeElement = <Account {...data} currentAddress="" />;
    const wrapper = shallow(customeElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly without associatedFund', () => {
    customeElement = <Account {...data} associatedFund="" />;
    const wrapper = shallow(customeElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly without networkId', () => {
    customeElement = <Account {...data} networkId="" />;
    const wrapper = shallow(customeElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick go to fund event', () => {
    const wrapper = shallow(defaultElement);
    wrapper
      .find('Button')
      .at(1)
      .simulate('click');
    expect(mockCallbackFund.mock.calls.length).toBe(1);
    expect(mockCallbackFund.mock.calls[0][0]).toEqual(
      '0x0890b03f83629B397F6E5E5075400fe6Ddb4255F',
    );
  });
});
