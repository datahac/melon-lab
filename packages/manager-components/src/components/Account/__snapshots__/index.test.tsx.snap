// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should render correctly 1`] = `
<div
  className="jsx-__scopedHash account"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Your Wallet
  </h3>
  <p
    className="jsx-__scopedHash"
  >
    Your ethereum address. Use this for white listing on
     
    <a
      className="jsx-__scopedHash"
      href="https://ico.bitcoinsuisse.ch/"
      target="_blank"
    >
      ico.bitcoinsuisse.ch
    </a>
    :
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x270c65cb31d037E4269dC8F736350FDc8ED07353"
        target="_blank"
      >
         
        0x270c65cb31d037E4269dC8F736350FDc8ED07353
         
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Associated fund address:
     
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x0890b03f83629B397F6E5E5075400fe6Ddb4255F"
        target="_blank"
      >
        0x0890b03f83629B397F6E5E5075400fe6Ddb4255F
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
      It is highly recommended to download a backup of your wallet. You can import this into
       
      <a
        className="jsx-__scopedHash"
        href="https://mycrypto.com/"
        target="_blank"
      >
        MyCrypto.com
      </a>
       
      or Parity.
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="secondary"
    >
      Download wallet backup JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="success"
    >
      Go to your fund
    </Button>
  </p>
  <br
    className="jsx-__scopedHash"
  />
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
       [IMPORTANT] - Please read carefully
    </strong>
     
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Careful, below actions have 
    <strong
      className="jsx-__scopedHash"
    >
       irreversible
    </strong>
     
    effects. If you do not have a backup of the mnemonic phrase that generated your current address,
    <strong
      className="jsx-__scopedHash"
    >
       
      you will never be able to access your current wallet again
       
    </strong>
     
    after performing one of the below actions.
  </p>
  <p
    className="jsx-__scopedHash"
  >
    If you do not wish to continue,
     
    <a
      className="jsx-__scopedHash"
      href="/"
    >
      click here to go back to your fund's page
    </a>
    .
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Create a new wallet
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Restore from mnemonic
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Import wallet JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Delete wallet
    </Button>
  </p>
</div>
`;

exports[`Account should render correctly without associatedFund 1`] = `
<div
  className="jsx-__scopedHash account"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Your Wallet
  </h3>
  <p
    className="jsx-__scopedHash"
  >
    Your ethereum address. Use this for white listing on
     
    <a
      className="jsx-__scopedHash"
      href="https://ico.bitcoinsuisse.ch/"
      target="_blank"
    >
      ico.bitcoinsuisse.ch
    </a>
    :
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x270c65cb31d037E4269dC8F736350FDc8ED07353"
        target="_blank"
      >
         
        0x270c65cb31d037E4269dC8F736350FDc8ED07353
         
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Associated fund address:
     
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x0890b03f83629B397F6E5E5075400fe6Ddb4255F"
        target="_blank"
      >
        0x0890b03f83629B397F6E5E5075400fe6Ddb4255F
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
      It is highly recommended to download a backup of your wallet. You can import this into
       
      <a
        className="jsx-__scopedHash"
        href="https://mycrypto.com/"
        target="_blank"
      >
        MyCrypto.com
      </a>
       
      or Parity.
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="secondary"
    >
      Download wallet backup JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="success"
    >
      Go to your fund
    </Button>
  </p>
  <br
    className="jsx-__scopedHash"
  />
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
       [IMPORTANT] - Please read carefully
    </strong>
     
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Careful, below actions have 
    <strong
      className="jsx-__scopedHash"
    >
       irreversible
    </strong>
     
    effects. If you do not have a backup of the mnemonic phrase that generated your current address,
    <strong
      className="jsx-__scopedHash"
    >
       
      you will never be able to access your current wallet again
       
    </strong>
     
    after performing one of the below actions.
  </p>
  <p
    className="jsx-__scopedHash"
  >
    If you do not wish to continue,
     
    <a
      className="jsx-__scopedHash"
      href="/"
    >
      click here to go back to your fund's page
    </a>
    .
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Create a new wallet
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Restore from mnemonic
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Import wallet JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Delete wallet
    </Button>
  </p>
</div>
`;

exports[`Account should render correctly without currentAddress 1`] = `
<div
  className="jsx-__scopedHash account"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Your Wallet
  </h3>
  <p
    className="jsx-__scopedHash"
  >
    Your ethereum address. Use this for white listing on
     
    <a
      className="jsx-__scopedHash"
      href="https://ico.bitcoinsuisse.ch/"
      target="_blank"
    >
      ico.bitcoinsuisse.ch
    </a>
    :
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x270c65cb31d037E4269dC8F736350FDc8ED07353"
        target="_blank"
      >
         
        0x270c65cb31d037E4269dC8F736350FDc8ED07353
         
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Associated fund address:
     
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x0890b03f83629B397F6E5E5075400fe6Ddb4255F"
        target="_blank"
      >
        0x0890b03f83629B397F6E5E5075400fe6Ddb4255F
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
      It is highly recommended to download a backup of your wallet. You can import this into
       
      <a
        className="jsx-__scopedHash"
        href="https://mycrypto.com/"
        target="_blank"
      >
        MyCrypto.com
      </a>
       
      or Parity.
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="secondary"
    >
      Download wallet backup JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="success"
    >
      Go to your fund
    </Button>
  </p>
  <br
    className="jsx-__scopedHash"
  />
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
       [IMPORTANT] - Please read carefully
    </strong>
     
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Careful, below actions have 
    <strong
      className="jsx-__scopedHash"
    >
       irreversible
    </strong>
     
    effects. If you do not have a backup of the mnemonic phrase that generated your current address,
    <strong
      className="jsx-__scopedHash"
    >
       
      you will never be able to access your current wallet again
       
    </strong>
     
    after performing one of the below actions.
  </p>
  <p
    className="jsx-__scopedHash"
  >
    If you do not wish to continue,
     
    <a
      className="jsx-__scopedHash"
      href="/"
    >
      click here to go back to your fund's page
    </a>
    .
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Create a new wallet
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Restore from mnemonic
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Import wallet JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Delete wallet
    </Button>
  </p>
</div>
`;

exports[`Account should render correctly without networkId 1`] = `
<div
  className="jsx-__scopedHash account"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Your Wallet
  </h3>
  <p
    className="jsx-__scopedHash"
  >
    Your ethereum address. Use this for white listing on
     
    <a
      className="jsx-__scopedHash"
      href="https://ico.bitcoinsuisse.ch/"
      target="_blank"
    >
      ico.bitcoinsuisse.ch
    </a>
    :
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x270c65cb31d037E4269dC8F736350FDc8ED07353"
        target="_blank"
      >
         
        0x270c65cb31d037E4269dC8F736350FDc8ED07353
         
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Associated fund address:
     
    <strong
      className="jsx-__scopedHash"
    >
      <a
        className="jsx-__scopedHash"
        href="https://kovan.etherscan.io/address/0x0890b03f83629B397F6E5E5075400fe6Ddb4255F"
        target="_blank"
      >
        0x0890b03f83629B397F6E5E5075400fe6Ddb4255F
      </a>
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
      It is highly recommended to download a backup of your wallet. You can import this into
       
      <a
        className="jsx-__scopedHash"
        href="https://mycrypto.com/"
        target="_blank"
      >
        MyCrypto.com
      </a>
       
      or Parity.
    </strong>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="secondary"
    >
      Download wallet backup JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="success"
    >
      Go to your fund
    </Button>
  </p>
  <br
    className="jsx-__scopedHash"
  />
  <p
    className="jsx-__scopedHash"
  >
    <strong
      className="jsx-__scopedHash"
    >
       [IMPORTANT] - Please read carefully
    </strong>
     
  </p>
  <p
    className="jsx-__scopedHash"
  >
    Careful, below actions have 
    <strong
      className="jsx-__scopedHash"
    >
       irreversible
    </strong>
     
    effects. If you do not have a backup of the mnemonic phrase that generated your current address,
    <strong
      className="jsx-__scopedHash"
    >
       
      you will never be able to access your current wallet again
       
    </strong>
     
    after performing one of the below actions.
  </p>
  <p
    className="jsx-__scopedHash"
  >
    If you do not wish to continue,
     
    <a
      className="jsx-__scopedHash"
      href="/"
    >
      click here to go back to your fund's page
    </a>
    .
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Create a new wallet
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Restore from mnemonic
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Import wallet JSON
    </Button>
  </p>
  <p
    className="jsx-__scopedHash"
  >
    <Button
      onClick={[Function]}
      style="danger"
    >
      Delete wallet
    </Button>
  </p>
</div>
`;
